dvc/stage/cache.py

```diff
@@ -250,12 +256,6 @@ def transfer(self, from_odb, to_odb):
                 rel = from_fs.path.as_posix(rel)
 
             dst = to_fs.path.join(to_odb.path, rel)
-            key = to_fs.path.parent(dst)
-            # check if any build cache already exists for this key
-            # TODO: check if MaxKeys=1 or something like that applies
-            # or otherwise this will take a lot of time!
-            if to_fs.exists(key) and first(to_fs.find(key)):
-                continue
```

Comment:
@daavoo This will affect both push and pull though, while `_load` that you've modified above is only affecting local run-cache entries. Changing this will mean that we will start uploading more stuff.

Reply:
>  Changing this will mean that we will start uploading more stuff.

Yes, but I thought it was the "lesser harm" for a simpler change. In the end, these are small files and I found these extra exists and find calls to potentially take more time than "blindly" push/pull all entries. 

If we want to keep the logic, I would need to replace the `find` with a full "listdir" and check for info (for modified time)

Reply:
But mtimes after download are not going to be preserved anyway, so pushing more stuff will harm here.

Reply:
I dropped the transfer changes to unblock this, "proper" solution would be something like https://github.com/iterative/dvc/pull/9908#issuecomment-1721191112 but this at least enables nuking and start with the new behavior

Reply:
@daavoo Seems like transfer changes are still here (or am I misunderstanding something?).

Reply:
forgot to push 😅